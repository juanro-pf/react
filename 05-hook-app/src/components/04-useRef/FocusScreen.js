import React, { useRef } from 'react';
import '../02-useEffect/effects.css';

export const FocusScreen = () => {

    const inputRef = useRef();
    
    const handleClick= () => {
        // document.querySelector('input').focus(); //.select hace focus + selecciona el texto
        inputRef.current.focus();
    };

    return (
        <div>
            <h1>Focus Screen</h1>
            <hr />

            <input 
                ref= {inputRef}
                className='form-control'
                placeholder='Su nombre'
            />

            <button
                className='btn btn-outline-primary mt-5'
                onClick= {handleClick}    
            >
                Focus
            </button>
        </div>
    );
};
