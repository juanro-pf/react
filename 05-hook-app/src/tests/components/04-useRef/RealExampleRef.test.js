import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import { RealExampleRef } from '../../../components/04-useRef/RealExampleRef';

describe('Testing RealExampleRef', () => {

    const wrapper= shallow(<RealExampleRef />);
    
    test('Snapshot', () => {
        
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('MultipleCustomHooks').exists()).toBe(false);

    });

    test('show MultipleCustomHooks', () => {

        wrapper.find('button').simulate('click');

        expect(wrapper.find('MultipleCustomHooks').exists()).toBe(true);

    });
    
});
