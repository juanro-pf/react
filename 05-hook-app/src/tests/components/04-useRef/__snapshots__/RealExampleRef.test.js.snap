// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing RealExampleRef Snapshot 1`] = `
<div>
  <h1>
    Real example ref
  </h1>
  <hr />
  <button
    className="btn btn-primary mt-5"
    onClick={[Function]}
  >
    Show / Hide
  </button>
</div>
`;
